feat: add Validation Strict Rules (#5445)
codeigniter4/CodeIgniter4: user_guide_src/source/libraries/validation.rst

If you submit the form you should see the success page or the form with error messages.

Config for Validation
*********************

.. _validation-traditional-and-strict-rules:

Traditional and Strict Rules
============================

> CI4 has two kind of Validation rule classes.

Reviewer: ```suggestion
CI4 has two kinds of Validation rule classes.
```

The default rule classes (**Traditional Rules**) have the namespace ``CodeIgniter\Validation``,
and the new classes (**Strict Rules**) have ``CodeIgniter\Validation\StrictRules``, which provide strict validation.

> The **Traditional Rules** implicitly assume that string values are validated.

Reviewer: What do you mean by "implicitly assume that string values are validated"? I think this needs to be rephrased.

Reviewer: You can also add a note that it uses PHP's type coercion.

Author: I mean the Traditional Rules implicitly assume to validate `$_POST`. 
The value type is only `string` or `string[]`.

> And the input value may be converted implicitly to a string value.

Reviewer: Do not start your sentence with And. You may add this to the previous sentence, or remove "and".

Author: Fixed.

It works for most basic cases like validating POST data.

> But, for example, if you use JSON input data, it may be type of bool/null/array.
> When you validate the bool ``true``, it is converted to string ``'1'`` with the Traditional rule classes.
> And if you validate it with the ``integer`` rule, ``'1'`` passes the validation.

Reviewer: ```suggestion
However, for example, if you use JSON input data, it may be a type of bool/null/array.
When you validate the boolean ``true``, it is converted to string ``'1'`` with the Traditional rule classes.
If you validate it with the ``integer`` rule, ``'1'`` passes the validation.
```

The **Strict Rules** don't use implicit type conversion.

.. warning:: When validating data that contains non-string values, such as JSON data, it is recommended to use **Strict Rules**.

Using Strict Rules
------------------

> If you want to use this, you need to change the rule classes in **app/Config/Validation.php**::

Reviewer: ```suggestion
If you want to use these rules, you need to change the rule classes in **app/Config/Validation.php**::
```
